modules/http/http-grammar.ym

```diff
@@ -109,6 +120,24 @@ http_options
     |
     ;
 
+on_error_action
+    : KW_SUCCESS { $$ = on_error_success; }
+    | KW_RETRY { $$ = on_error_retry; }
+    | KW_DROP { $$ = on_error_drop; }
+    | KW_DISCONNECT { $$ = on_error_disconnect; }
+    ;
+
+on_error_block
+    : '(' { last_on_error_params = g_new0(OnErrorParams, 1); }
+       positive_integer { last_on_error_params->status_code = $3; }
+       KW_ACTION '(' on_error_action ')' { last_on_error_params->action = $7; }
+      ')' {
+             CHECK_ERROR(last_on_error_params->action, @1, "action is mandatory parameter");
```

Comment:
I don't see how action would be unset here.

Reply:
Changed code to not to use allocation. Renamed add to insert.

Action will be unset by the hashtable.